diff -ru a/src/dhcp-manager/systemd-dhcp/nm-sd-adapt.h b/src/dhcp-manager/systemd-dhcp/nm-sd-adapt.h
--- a/src/dhcp-manager/systemd-dhcp/nm-sd-adapt.h       2016-04-01 23:27:52.000000000 +0200
+++ b/src/dhcp-manager/systemd-dhcp/nm-sd-adapt.h       2021-01-21 11:39:22.813379325 +0100
@@ -124,9 +124,11 @@
 #include "async.h"
 #include "util.h"
 
+#if !defined(_GNU_SOURCE) || !defined(__GLIBC__) || __GLIBC__ < 2 || (__GLIBC__ == 2 && __GLIBC_MINOR__ < 30)
 static inline pid_t gettid(void) {
         return (pid_t) syscall(SYS_gettid);
 }
+#endif
 
 #endif /* NM_SD_ADAPT_H */
 

